bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,807 @@
+        } else if (unit.equals(HeliosVariable.UNIT_SEC)) {
+            return new QuantityType<>(Integer.parseInt(value), SmartHomeUnits.SECOND);
+        } else if (unit.equals(HeliosVariable.UNIT_VOLT)) {
+            return new QuantityType<>(Float.parseFloat(value), SmartHomeUnits.VOLT);
+        } else if (unit.equals(HeliosVariable.UNIT_PERCENT)) {
+            return new QuantityType<>(Float.parseFloat(value), SmartHomeUnits.PERCENT);
+        } else if (unit.equals(HeliosVariable.UNIT_PPM)) {
+            return new QuantityType<>(Float.parseFloat(value), SmartHomeUnits.PARTS_PER_MILLION);
+        } else if (unit.equals(HeliosVariable.UNIT_TEMP)) {
+            return new QuantityType<>(Float.parseFloat(value), SIUnits.CELSIUS);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Prepares the payload for the request
+     *
+     * @param payload The String representation of the payload
+     * @return The Register representation of the payload
+     */
+    private ModbusRegister[] preparePayload(String payload) {
+
+        // determine number of registers
+        int l = (payload.length() + 1) / 2; // +1 because we need to include at least one termination symbol 0x00
+        if ((payload.length() + 1) % 2 != 0) {
+            l++;
+        }
+
+        ModbusRegister reg[] = new ModbusRegister[l];
+        byte[] b = payload.getBytes();
+        int ch = 0;
+        for (int i = 0; i < reg.length; i++) {
+            byte b1 = ch < b.length ? b[ch] : (byte) 0x00; // terminate with 0x00 if at the end of the payload
+            ch++;
+            byte b2 = ch < b.length ? b[ch] : (byte) 0x00;
+            ch++;
+            reg[i] = new ModbusRegister(b1, b2);
+        }
+        return reg;
+    }
+
+    /**
+     * Decodes the Helios device' response and updates the channel with the actual value of the variable
+     *
+     * @param response The registers received from the Helios device
+     * @return The value or <tt>null</tt> if an error occurred
+     */
+    private void processResponse(HeliosVariable v, ModbusRegisterArray registers) {
+        // decode response
+        byte[] b = new byte[registers.size() * 2];
+        int actSize = 0; // track the actual size of the usable array (excluding any 0x00
+                         // characters)
+        for (int i = 0; i < registers.size(); i++) {
+            byte[] reg = registers.getRegister(i).getBytes();
+            if (reg.length == 2) { // only add to the array if it's a usable character
+                if (reg[0] != 0x00) {
+                    b[actSize++] = reg[0];
+                }
+                if (reg[1] != 0x00) {
+                    b[actSize++] = reg[1];
+                }
+            }
+        }
+        b = Arrays.copyOf(b, actSize); // before creating a string of it the array needs to be
+                                       // truncated
+        String r = new String(b, StandardCharsets.US_ASCII);
```

Comment:
Would this work?

```suggestion
        String r = ModbusBitUtilities
                .extractStringFromRegisters(registers, 0, registers.size() * 2, StandardCharsets.US_ASCII).toString();
```

Reply:
Yes, it works nicely - thanks for the hint!